Legacy/bonej/src/main/java/org/bonej/plugins/ParticleAnalysis.java

```diff
@@ -717,42 +719,62 @@ private static double getSurfaceArea(final List<Point3f> points) {
 	 * Get the Feret diameter of a surface. Uses an inefficient brute-force
 	 * algorithm.
 	 *
-	 * @param surfacePoints points of all the particles.
-	 * @return Feret diameters of the surfaces.
+	 * @param surfacePoints points from a surface mesher
+	 * @throws IllegalArgumentException if there are less than 12 points (a tetrahedron)
+	 * @return Feret diameters and x, y, z coordinates of the two feret points of each surface,
+	 * packed in a double so that the feret diameter of particle i is found at [i * 7] and the
+	 * points' coordinates are in the following 6 positions in ax, ay, az, bx, by, bz order.
 	 */
-	static double[] getFerets(final List<List<Point3f>> surfacePoints) {
+	static double[][] getFerets(final List<List<Point3f>> surfacePoints) {
 		Thread[] threads = Multithreader.newThreads();
 		final int nSurfaces = surfacePoints.size();
-		final double[] ferets = new double[nSurfaces];
+		//distance, xa, ya, za, xb, yb, zb
+		final double[][] ferets = new double[nSurfaces][7];
 		AtomicInteger ai = new AtomicInteger(0);
 		for (int thread = 0; thread < threads.length; thread++) {
 			threads[thread] = new Thread(() -> {
 				for (int i = ai.getAndIncrement(); i < nSurfaces; i = ai.getAndIncrement()) {
 					final List<Point3f> surface = surfacePoints.get(i);
-				
+					
+					
 					if (surface == null) {
-						ferets[i] = Double.NaN;
+						Arrays.fill(ferets[i], Double.NaN);
 						continue;
 					}
 					
+					final int nPoints = surface.size();
+					
+					//4 triangles * 3 points for the minimal tetrahedron
+					if (nPoints < 12) {
+						Arrays.fill(ferets[i], Double.NaN);
+						throw new IllegalArgumentException("Mesh for particle "+i+" has too few points: "+nPoints);
```

Comment:
Since you're filling the array with NaN values, it seems that a surface that's not a tetrahedron is not a show stopper. In that case let's not throw an exception, but just
```
// Since behaviour for null and empty lists is the same, let's not have 'null' lists in `surfacePoints`
if (surface == null || surfacesurface.size() < 12) {
  Arrays.fill(ferets[i], Double.NaN);
  // Can still log a strange surface here
}
```

If finding a surface with less than 12 points is truly a bug that should stop execution and throw an exception then you should see if there is one at the beginning of the method:
```
// filter(Objects::nonNull) not necessary if there are no nulls in surfacePoints
final boolean strangeSurfaces = surfacePoints.stream().filter(Objects::nonNull).anyMatch(surface -> surface.size() < 12);
if (strangeSurfaces) {
    throw new IllegalArgumentException("Bad surface!");
}
```
As the code is now only the threads that throw exceptions stop, while the rest chug happily along